index.js

```diff
@@ -238,25 +238,24 @@ async function checkHeartbeat() {
   }
 }
 
-async function handleSensorRequest(currentLocationId, radarType) {
+async function handleSensorRequest(location, radarType) {
   let statemachine
   if (radarType === RADAR_TYPE.XETHRU) {
-    statemachine = new XeThruStateMachine(currentLocationId)
+    statemachine = new XeThruStateMachine(location.locationid)
   } else if (radarType === RADAR_TYPE.INNOSENT) {
-    statemachine = new InnosentStateMachine(currentLocationId)
+    statemachine = new InnosentStateMachine(location.locationid)
   }
   const currentState = await statemachine.getNextState(db, redis)
-  const stateobject = await redis.getLatestLocationStatesData(currentLocationId)
+  const stateobject = await redis.getLatestLocationStatesData(location.locationid)
   let prevState
   if (!stateobject) {
     prevState = STATE.RESET
   } else {
     prevState = stateobject.state
   }
-  const location = await db.getLocationData(currentLocationId)
```

Comment:
Moved this outside of the function `handleSensorRequest` and made it the responsibility of the caller to get the Location and pass it in because that way the caller could check whether the Location was active and only call this function if it is. Also, two out of the three places where `handleSensorRequest` was being called already had the whole Location object, so this is more efficient to not have this second DB call.